gofish/dell/networkDeviceFunction.go

```diff
@@ -0,0 +1,364 @@
+	FlexAddressingSupport            string
+	IPSecOffloadSupport              string
+	MACSecSupport                    string
+	NWManagementPassThrough          string
+	NicPartitioningSupport           string
+	OSBMCManagementPassThrough       string
+	OnChipThermalSensor              string
+	OpenFlowSupport                  string
+	PXEBootSupport                   string
+	PartitionWOLSupport              string
+	PersistencePolicySupport         string
+	PriorityFlowControl              string
+	RDMASupport                      string
+	RemotePHY                        string
+	TCPChimneySupport                string
+	TCPOffloadEngineSupport          string
+	VEB                              string
+	VEBVEPAMultiChannel              string
+	VEBVEPASingleChannel             string
+	VirtualLinkControl               string
+	ISCSIBootSupport                 string
+	ISCSIOffloadSupport              string
+	UEFISupport                      string
+}
+
+// FCCapabilities holds the json model of Dell OEM data for DellFCCapabilities.
+type FCCapabilities struct {
+	Entity
+	FCMaxNumberExchanges           int
+	FCMaxNumberOutStandingCommands int
+	FeatureLicensingSupport        string
+	FlexAddressingSupport          string
+	OnChipThermalSensor            string
+	PersistencePolicySupport       string
+	UEFISupport                    string
+}
+
+// NetworkDeviceFunctionExtended contains gofish NetworkDeviceFunction data,
+// as well as Dell OEM data, DellEthernet, PhysicalPortAssignment, AssignablePhysicalPorts and AssignablePhysicalNetworkPorts.
+type NetworkDeviceFunctionExtended struct {
+	*redfish.NetworkDeviceFunction
+	OemData                            NetworkDeviceFunctionOEM
+	DellAssignablePhysicalPorts        []Entity
+	DellAssignablePhysicalNetworkPorts []Entity
+	DellPhysicalPortAssignment         Entity
+	DellEthernet                       Ethernet
+}
+
+// Ethernet is the json model of Ethernet including VLAN data.
+type Ethernet struct {
+	MACAddress          string
+	MTUSize             int
+	PermanentMACAddress string
+	VLAN                VLAN
+}
+
+// VLAN is the json model of VLAN
+type VLAN struct {
+	VLANEnabled bool
+	VLANID      int
+}
+
+// NetworkDeviceFunction returns a Dell.NetworkDeviceFunction pointer given a redfish.NetworkDeviceFunction pointer from Gofish.
+// This is the wrapper that extracts and parses Dell NetworkDeviceFunction OEM data,
+// as well as DellAssignablePhysicalPorts, DellAssignablePhysicalNetworkPorts, DellPhysicalPortAssignment and DellEthernet.
+func NetworkDeviceFunction(deviceFunction *redfish.NetworkDeviceFunction) (*NetworkDeviceFunctionExtended, error) {
+	dellNetworkDeviceFunction := &NetworkDeviceFunctionExtended{NetworkDeviceFunction: deviceFunction}
+
+	rawDataBytes, err := GetRawDataBytes(deviceFunction)
+	if err != nil {
+		return dellNetworkDeviceFunction, err
+	}
+	if oemRawData, found := GetNodeFromRawDataBytes(rawDataBytes, "Oem"); found == nil {
```

Comment:
Many `GetNodeFromRawDataBytes`. Maybe can be refactored.

Reply:
thanks @baoy1 , will do that

Reply:
`*redfish.NetworkDeviceFunction` type has a public field `Oem`. You can directly use that.

Reply:
hi @rounak-adhikary , for current gofish commit version, the NetworkDeviceFunction only has private rawData, no public  Oem. But in the latest gofish , *redfish.NetworkDeviceFunction type has a public field Oem.

Reply:
Okay, thanks!